File: python/diamond/diamond.py
A: int = ord("A")


def rows(which: str) -> list[str]:
    """Port of ../../lua/diamond/diamond.lua"""
    letter = ord(which) - A + 1  # letter is 1 for A, 2 for B, 3 for C
    lines = [" " * (letter - 1) + "A" + " " * (letter - 1)]
    for position in range(2, letter + 1):  # add one as stop for range is exclusive (1)
        char = chr(A + position - 1)
        lines.append(  # append to list, rather than use sparse table (2)
            "".join(
                (
                    " " * (letter - position),
                    char,
                    " " * (2 * position - 3),
                    char,
                    " " * (letter - position),
                )
            )
        )
    for i in range(len(lines) + 1, 2 * letter):
        lines.append(
            lines[2 * letter - i - 1]  # subtract 1 as Python lists are zero based (3)
        )
    return lines

File: lua/diamond/diamond.lua
A = utf8.codepoint("A")
return function(which)
	local letter = utf8.codepoint(which) - A + 1 -- letter is 1 for A, 2 for B, 3 for C
	local lines = { string.rep(" ", letter - 1) .. "A" .. string.rep(" ", letter - 1) }
	for position = 2, letter do
		local char = utf8.char(A + position - 1)
		lines[position] = table.concat({
			string.rep(" ", letter - position),
			char,
			string.rep(" ", 2 * position - 3),
			char,
			string.rep(" ", letter - position),
		})
	end
	for i = #lines + 1, 2 * letter - 1 do
		lines[i] = lines[2 * letter - i]
	end
	return table.concat(lines, "\n") .. "\n"
end
